.github/actions/cache-query-compilation/move-caches.js

```diff
@@ -0,0 +1,80 @@
+// # Move all the existing cache into another folder, so we only preserve the cache for the current queries.
+// mkdir -p ${COMBINED_CACHE_DIR}
+// rm -f **/.cache/{lock,size} # -f to avoid errors if the cache is empty.
+// # copy the contents of the .cache folders into the combined cache folder.
+// cp -r **/.cache/* ${COMBINED_CACHE_DIR}/ || : # ignore missing files
+// # clean up the .cache folders
+// rm -rf **/.cache/*
+
+const fs = require("fs");
+const path = require("path");
+
+// the first argv is the cache folder to create.
+const COMBINED_CACHE_DIR = process.argv[2];
+
+function* walkCaches(dir) {
+  const files = fs.readdirSync(dir);
+  for (const file of files) {
+    const filePath = path.join(dir, file);
+    if (!fs.existsSync(filePath)) {
+      continue;
+    }
+    const stat = fs.statSync(filePath);
+    if (stat.isDirectory()) {
+      yield* walkCaches(filePath);
+      if (file === ".cache") {
+        yield filePath;
+      }
+    }
+  }
+}
+
+async function copyDir(src, dest) {
```

Comment:
Since node 16.7.0 the following is possible:  `fs.cp(src, dest, {recursive: true});` . Not sure if that is available on Actions, it's also marked as "experimental".

Reply:
I wouldn't trust that to be available yet.